r/tests/testthat/test-compute-string-split.R

```diff
@@ -0,0 +1,32 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+test_str <-c("foobar, foo, bar", "baz, qux, quux.")
+
+test_that("strsplit for Array and ChunkedArray ", {
+  expect_vector_equal(strsplit(input, split = ","), test_str, ignore_attr = TRUE)
+})
+
+test_that("strsplit for Scalar", {
+  test_scalar <- Scalar$create(test_str[1])
+  expect_equal(
+    strsplit(test_scalar, split = ","),
+    Scalar$create(list(c("foobar", " foo", " bar")))
+  )
+})
```

Comment:
Given the discussion above about masking, I think you could modify the tests here to just use `call_function("split_pattern", ...)` instead of `strsplit()`. That provides a direct test that the code you added in `compute.cpp` is working as expected. You could also test `max_splits` and `reverse` that way.

And to test `strsplit()` inside dplyr verbs, you can add additional test code in `test-dplyr-string-functions.R`

Reply:
I don't think you need to test `call_function("split_pattern", ...)` here, that's covered in C++ tests. I'd just test in dplyr, following the sub/gsub model where appropriate.

Reply:
The only catch is that the dplyr tests won't exercise the `max_splits` and `reverse` arguments, because `strsplit()` doesn't use either.

The stringr splitting functions take an `n` argument that *might* exercise `max_splits` (but I'm not sure)

Reply:
You would test that in dplyr by calling `arrow_spilt_pattern` with the extra options

Reply:
Oh, right, I forgot about that. Thanks

Reply:
Removed test now